### ISSUE TITLE
Some emojis from the preview stay on the screen after closing file selector.

### ISSUE THREAD

**Issue by MGlolenstine** [helix-editor/helix#5997]
### Summary

When looking at a file that contains emojis within the file selector ([example](https://github.com/woodemi/quick.flutter/tree/master/packages/quick_blue)), the emojis remain visible even after the selector is closed and don't vanish until overwritten with different content.

![image](https://user-images.githubusercontent.com/9406770/218958753-9a88e6d1-f59e-4719-964e-cc65284eae9c.png)

EDIT:
Checkmarks seem to stay even if we're just scrolling the content.

### Reproduction Steps

 

I tried this:

1. `hx`
2. Paste in the following
```
| API | Android | iOS | macOS | Windows | Linux |
| :--- | :---: | :---: | :---: | :---: | :---: |
| availabilityChangeStream | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| isBluetoothAvailable | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| startScan/stopScan | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| connect/disconnect | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| discoverServices | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| setNotifiable | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| readValue | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| writeValue | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ |
| requestMtu | ✔️ | ✔️ | ✔️ | ✔️ |  |
```
3. Open the file or buffer picker (SPACE+F or SPACE+B) and select the current file.
4. Close the picker.

I expected this to happen:
Clean screen

Instead, this happened:
Some remains of the preview remain on screen as can be seen on the above picture.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

I've opened the `hx`, and loaded the `a` file that contained the above check-marked text.

```
2023-02-15T08:22:50.406 helix_loader [DEBUG] Located configuration folders: []
2023-02-15T08:22:50.413 helix_view::clipboard [DEBUG] Using xclip to interact with the system and selection (primary) clipboard
2023-02-15T08:22:50.414 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-02-15T08:22:50.815 helix_term::application [DEBUG] received editor event: IdleTimer
2023-02-15T08:22:54.328 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-02-15T08:22:54.552 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-02-15T08:22:54.729 helix_term::application [DEBUG] received editor event: IdleTimer
2023-02-15T08:22:55.088 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-02-15T08:22:55.448 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-02-15T08:22:56.464 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-02-15T08:22:56.464 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:57.649 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:57.809 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:57.809 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:58.210 helix_term::application [DEBUG] received editor event: IdleTimer
2023-02-15T08:22:58.211 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:59.113 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:59.473 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:22:59.874 helix_term::application [DEBUG] received editor event: IdleTimer
2023-02-15T08:23:01.345 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:23:01.585 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:23:01.649 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:23:01.747 helix_term::application [DEBUG] received editor event: IdleTimer
2023-02-15T08:23:01.833 helix_view::document [DEBUG] submitting save of doc 'Some("/home/arch/tmp/test/a")'
2023-02-15T08:23:01.833 helix_term::job [DEBUG] waiting on jobs...
2023-02-15T08:23:01.833 helix_view::document [DEBUG] doc 2 revision updated 0 -> 0
2023-02-15T08:23:01.833 helix_term::commands::typed [DEBUG] quitting...
2023-02-15T08:23:01.833 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-02-15T08:23:01.833 helix_term::job [DEBUG] waiting on jobs...
2023-02-15T08:23:01.833 helix_term::job [DEBUG] waiting on jobs...
```

</details>

### Platform

Linux

### Terminal Emulator

kitty 0.26.5

### Helix Version

helix 22.12

**Comment by pokkos** [comment]
I can reproduce this issue. You can also see it if you step the cursor along the line that the characters get shifted. Or when selecting multiple lines that the lines get shorter.

This seems to be a Grapheme handling issue, also related to #4932 and #5502.

**Comment by pascalkuthe** [comment]
Closing in favor of https://github.com/helix-editor/helix/issues/6012
